### ISSUE TITLE
Implementing new Material 3(alpha) components in Angular project

### ISSUE THREAD

**Issue by neo-xy** [material-components/material-web#3826]
I want to test  `@material/web@0.1.0-alpha.0` but failing on implementing it.

What is the simplest way to implement Material 3 web-components in Angular project?

I found this article : [https://sam-redmond.com/using-web-components-in-angular-d56a541ee2f1](url)
but seems like a lot of a hustle..:/

**Comment by BalajiDany** [comment]
@neo-xy , did you found any way to implement?

**Comment by neo-xy** [comment]
@BalajiDany unfortunately not :(

**Comment by BrentonJScott** [comment]
Try to follow this [ tutorial ](https://levelup.gitconnected.com/how-to-use-web-components-in-angular-c5d1d7825ca0)

You can add the components in a module and add the components to the bootstrap array. Also don't forget to add this **schemas: [ CUSTOM_ELEMENTS_SCHEMA ]**


`import { MdTextButton } from '@material/web/button/text-button';`
`import { MdFilledButton } from '@material/web/button/filled-button';`

`@NgModule({`
`  declarations: [`
`    DashLayoutComponent,`
`  ],`
`  imports: [`
`    DashRoutingModule,`
`    CommonModule`
`  ],`
`  schemas: [ CUSTOM_ELEMENTS_SCHEMA ],`
`  bootstrap: [`
`    MdTextButton,`
`    MdFilledButton,`
`  ]`
`})`
`export class DashModule { }`

**Comment by neo-xy** [comment]
@BrentonJScott  thank. you very much ,sadly getting
"Class MdFilledButton is neither an Angular component nor directive nor pipe " in app nodule in "declorations" 
Google it up and found some solutions that I tryed (setting aoe:false, checking in recoompile for typescript, Invalidate caches... , adding: customElements.define('md-text-button', MdFilledButton) to module) but none of them worked

In tutorial that U linked it says that components should be listed in "declarations" althoug in your exampel you list those in "boostrap" either way I tested both and same error comming up. 

Angular 15.0.4
node 18.10.0
typescript 4.8.4

If you have any suggestion I would very so appreciate those:)

![Skärmavbild 2023-04-07 kl  11 26 54](https://user-images.githubusercontent.com/17005270/230583995-56f85983-8638-426f-9bdf-cef270a7e8d2.png)

**Comment by tanishqmanuja** [comment]
Maybe something like this,

```typescript
import { Component, ElementRef, Input, NgZone, inject } from "@angular/core";
import type { MdDivider as MdDividerElement } from "@material/web/divider/divider.js";
import "@material/web/divider/divider.js";
import {
	BooleanInput,
	coerceBooleanProperty,
} from "src/app/shared/utils/coercion/boolean-property";

@Component({
	selector: "md-divider",
	standalone: true,
	template: ` <ng-content></ng-content>`,
})
export class MdDividerComponent {
	private el: MdDividerElement = inject(ElementRef).nativeElement;
	private ngZone = inject(NgZone);

	@Input()
	set inset(v: BooleanInput) {
		this.ngZone.runOutsideAngular(() => {
			this.el.inset = coerceBooleanProperty(v);
		});
	}
	get inset() {
		return this.el.inset;
	}

	@Input()
	set insetStart(v: BooleanInput) {
		this.ngZone.runOutsideAngular(() => {
			this.el.insetStart = coerceBooleanProperty(v);
		});
	}
	get insetStart() {
		return this.el.insetStart;
	}

	@Input()
	set insetEnd(v: BooleanInput) {
		this.ngZone.runOutsideAngular(() => {
			this.el.insetEnd = coerceBooleanProperty(v);
		});
	}
	get insetEnd() {
		return this.el.insetEnd;
	}
}

```

I am not sure it is the best way to wrap lit components. But it works for now and also provide typescript completions in vscode.

**Comment by BrentonJScott** [comment]
@neo-xy did you add custom schema?

Like this

<img width="688" alt="image" src="https://user-images.githubusercontent.com/103521333/237031235-c008e02c-61c2-46c6-be1c-4d29ad8d8970.png">

**Comment by tanishqmanuja** [comment]
@BrentonJScott, Would be really helpful if you can also provide a snippet for standalone app

**Comment by neo-xy** [comment]
@BrentonJScott yes, I did :/

![Skärmavbild 2023-05-14 kl  08 30 11](https://github.com/material-components/material-web/assets/17005270/96d2eb1a-8550-4cb6-874b-a5f46ceac247)

**Comment by neo-xy** [comment]
I turns out that the issue was intelliJ - latest version fixes the issue (2023.2)
